servicex/resources/submit_transformation_request.py

```diff
@@ -39,64 +40,92 @@
 parser.add_argument('workers', required=False, type=int)
 parser.add_argument('result-destination', required=True, choices=['kafka', 'object-store'])
 parser.add_argument('result-format', required=False,
-                    choices=['arrow', 'parquet'], default='arrow')
+                    choices=['arrow', 'parquet', 'root-file'], default='arrow')
 parser.add_argument('kafka', required=False, type=dict)
 
 kafka_parser = reqparse.RequestParser()
 kafka_parser.add_argument('broker', required=False, location=('kafka'))
 
 
+def _workflow_name(transform_request):
+    'Look at the keys and determine what sort of a workflow we want to run'
+    has_columns = ('columns' in transform_request) and transform_request['columns']
+    has_selection = ('selection' in transform_request) and transform_request['selection']
+    if has_columns and not has_selection:
+        return 'straight_transform'
+    if not has_columns and has_selection:
+        return 'selection_codegen'
+    raise ValueError('Cannot determine workflow from argument - '
+                     'selection or columns must be given, and not both')
+
+
 class SubmitTransformationRequest(ServiceXResource):
     @classmethod
-    def make_api(cls, rabbitmq_adaptor, object_store, elasticsearch_adapter):
+    def make_api(cls, rabbitmq_adaptor, object_store, elasticsearch_adapter, code_gen_service):
         cls.rabbitmq_adaptor = rabbitmq_adaptor
         cls.object_store = object_store
         cls.elasticsearch_adapter = elasticsearch_adapter
+        cls.code_gen_service = code_gen_service
         return cls
 
     def post(self):
-        transformation_request = parser.parse_args()
-        request_id = str(uuid.uuid4())
-        time = datetime.now(tz=timezone.utc)
-
-        if self.object_store and \
-                transformation_request['result-destination'] == 'object-store':
-            self.object_store.create_bucket(request_id)
-
-        if transformation_request['result-destination'] == 'kafka':
-            broker = transformation_request['kafka']['broker']
-        else:
-            broker = None
-
-        request_rec = TransformRequest(
-            did=transformation_request['did'],
-            submit_time=time,
-            columns=transformation_request['columns'],
-            request_id=str(request_id),
-            image=transformation_request['image'],
-            chunk_size=transformation_request['chunk-size'],
-            result_destination=transformation_request['result-destination'],
-            result_format=transformation_request['result-format'],
-            kafka_broker=broker,
-            workers=transformation_request['workers']
-        )
-
-        did_request = {
-            "request_id": request_rec.request_id,
-            "did": request_rec.did,
-            "service-endpoint": self._generate_advertised_endpoint(
-                "servicex/transformation/" +
-                request_rec.request_id
+        try:
+            try:
```

Comment:
I like the nested try statements here. Is there any value to adding more? Maybe individual tries in request_rec so if a request failed to assign a necessary value you'd get something more tailored than "Something went wrong"?

Reply:
Actually, the nested try statement can be removed and replaced with the correct `werkzeug.exceptions.BadRequest` in the outer try block. I've pushed this change.
I agree that we need some more fine grain exceptions down in the list of exceptions. For example I bet there are some specific exceptions from RabbitMQ that we could trap.
Thanks for pointing this out - I've created issue #13 so we can dig into this later.